### ISSUE TITLE
ESP32-CAM - Ability to disable status LED during compile

### ISSUE THREAD

**Issue by darranwil** [forkineye/ESPixelStick#405]
ESP32-CAM - Ability to disable status LED during compile.

**Describe the solution you'd like**
Option in source to disable status LED on ESP32-cam board. Nice to see it's working but super bright and puts out a ton of heat.

**Additional context**

Thank you again for adding ESP32-Cam support!  Much love and respect to the Devs.

**Comment by darranwil** [comment]
Man you guys are fast! Just walked in the door. Yup the flash LED.

**Comment by darranwil** [comment]
Downloaded latest code last night and flashed. Flash LED still turns on.

**Comment by forkineye** [comment]
> Downloaded latest code last night and flashed. Flash LED still turns on.

Can you link the specific board you have? I have a feeling there's a few different versions of these.

**Comment by darranwil** [comment]
It's an AI-Thinker clone. I use the same boards with ESPHome and it has GPIO 4 defined as the flash LED pin.  Maybe I compiled it wrong, let me try again and report back.

Here is a good list of the common ESP32-Cam boards used in ESPHome: https://esphome.io/components/esp32_camera.html

**Comment by forkineye** [comment]
Make sure you're choosing the esp32cam environment in pio if compiling yourself. The prebuilt CI releases under actions should work as well.

**Comment by thomastech** [comment]
As was mentioned earlier, the ESP32CAM build sets GPIO4 to an input with the pullup enabled. I suspect the pullup is allowing the LED drive transistor to turn on. 

As an experiment, find this line in OutputMgr.cpp:
`pinMode (LED_GPIO, INPUT_PULLUP);`

And change to:
`pinMode (LED_GPIO, INPUT);`

Or maybe setting GPIO4 to the input configuration is not the ideal way to handle the LED drive pin. I suspect there may be a backstory to doing it this way in ESPixelStick, would be interesting to know the reason for it. 

So if no joy, then confirm GPIO4 is involved by changing the code above to this:
`pinMode (LED_GPIO, OUTPUT);
digitalWrite (LED_GPIO, LOW)`

It may be possible to leave everything as-is and temporarily solved it with the UI's Device Setup page. Configure the Relay Output function and set a GPIO ID to 4. Test it with the "Inverted" selection disabled. Just thinking out loud here, no promises.

- Thomas

**Comment by forkineye** [comment]
I worked without the explicit configuration for me when testing. It was added in attempt to resolve @darranwil's issue, however I'm unsure if he was using the d1_mini32 or esp32_cam environment when the issue was created.

**Comment by forkineye** [comment]
Seems to be working, closing due to inactivity.
